Para esse projeto, foi selecionado o DataFrame 'healthcare_dataset_stroke_data.csv' encontrado no Kaggle.com, com a finalidade de identificar topicos e responder as perguntas propostas do desafio

In [0]:
from pyspark.sql import SparkSession
#importa a biblioteca que cria a seção do spark

spark = SparkSession.builder \
                    .appName("desafio_IGTI") \
                    .getOrCreate() 
#cria a seção caso não exista ou obtém a já criada

In [0]:
diretorio_dataset="/FileStore/tables/healthcare_dataset_stroke_data.csv"  
#diretório que contém o arquivo a ser utilizado

df = spark.read.format("csv") \
                .options(header="true", inferschema="true") \
                .load(diretorio_dataset)  
#realiza a leitura do dataset

In [0]:
df.show()

+-----+------+----+------------+-------------+------------+-------------+--------------+-----------------+----+---------------+------+
|   id|gender| age|hypertension|heart_disease|ever_married|    work_type|Residence_type|avg_glucose_level| bmi| smoking_status|stroke|
+-----+------+----+------------+-------------+------------+-------------+--------------+-----------------+----+---------------+------+
| 9046|  Male|67.0|           0|            1|         Yes|      Private|         Urban|           228.69|36.6|formerly smoked|     1|
|51676|Female|61.0|           0|            0|         Yes|Self-employed|         Rural|           202.21| N/A|   never smoked|     1|
|31112|  Male|80.0|           0|            1|         Yes|      Private|         Rural|           105.92|32.5|   never smoked|     1|
|60182|Female|49.0|           0|            0|         Yes|      Private|         Urban|           171.23|34.4|         smokes|     1|
| 1665|Female|79.0|           1|            0|         

In [0]:
df.printSchema()

root
 |-- id: integer (nullable = true)
 |-- gender: string (nullable = true)
 |-- age: double (nullable = true)
 |-- hypertension: integer (nullable = true)
 |-- heart_disease: integer (nullable = true)
 |-- ever_married: string (nullable = true)
 |-- work_type: string (nullable = true)
 |-- Residence_type: string (nullable = true)
 |-- avg_glucose_level: double (nullable = true)
 |-- bmi: string (nullable = true)
 |-- smoking_status: string (nullable = true)
 |-- stroke: integer (nullable = true)



In [0]:
df.describe().show()

+-------+-----------------+------+------------------+------------------+-------------------+------------+---------+--------------+------------------+------------------+--------------+-------------------+
|summary|               id|gender|               age|      hypertension|      heart_disease|ever_married|work_type|Residence_type| avg_glucose_level|               bmi|smoking_status|             stroke|
+-------+-----------------+------+------------------+------------------+-------------------+------------+---------+--------------+------------------+------------------+--------------+-------------------+
|  count|             5110|  5110|              5110|              5110|               5110|        5110|     5110|          5110|              5110|              5110|          5110|               5110|
|   mean|36517.82935420744|  null|43.226614481409015|0.0974559686888454|0.05401174168297456|        null|     null|          null|106.14767710371804|28.893236911794673|          null| 

In [0]:
print(f' Quantidade de colunas existentes :  {len(df.columns)}')
print(f' Quantidade de linhas existentes :  {df.count()}')

 Quantidade de colunas existentes :  12
 Quantidade de linhas existentes :  5110


In [0]:
df.printSchema()

root
 |-- id: integer (nullable = true)
 |-- gender: string (nullable = true)
 |-- age: double (nullable = true)
 |-- hypertension: integer (nullable = true)
 |-- heart_disease: integer (nullable = true)
 |-- ever_married: string (nullable = true)
 |-- work_type: string (nullable = true)
 |-- Residence_type: string (nullable = true)
 |-- avg_glucose_level: double (nullable = true)
 |-- bmi: string (nullable = true)
 |-- smoking_status: string (nullable = true)
 |-- stroke: integer (nullable = true)



A partir da verificacao conclui-se que existem 6 variaveis tipo STRING

In [0]:
df.select('age').describe().show()

+-------+------------------+
|summary|               age|
+-------+------------------+
|  count|              5110|
|   mean|43.226614481409015|
| stddev| 22.61264672311348|
|    min|              0.08|
|    max|              82.0|
+-------+------------------+



In [0]:
display(df.groupby('gender','stroke').count().sort("count",ascending=True))

gender stroke count Other 0 1 Male 1 108 Female 1 141 Male 0 2007 Female 0 2853

In [0]:
# verificando a existencia de dados nulos
pandas = df.toPandas()
pandas.isnull().sum()

Out[37]: id                   0
gender               0
age                  0
hypertension         0
heart_disease        0
ever_married         0
work_type            0
Residence_type       0
avg_glucose_level    0
bmi                  0
smoking_status       0
stroke               0
dtype: int64

In [0]:
# verificando se a variavel BMi possui valor nao numerico
df.select('bmi').dtypes

Out[38]: [('bmi', 'string')]

In [0]:
# Verificando a existencia de duas classes de residencia na variavel
df.groupby('Residence_type').count().show()

+--------------+-----+
|Residence_type|count|
+--------------+-----+
|         Urban| 2596|
|         Rural| 2514|
+--------------+-----+



In [0]:
# verificando se existe balanceamento entre individuoes que sofreram ou nao AVC
df.groupby('stroke').count().show()

+------+-----+
|stroke|count|
+------+-----+
|     1|  249|
|     0| 4861|
+------+-----+



Unica resposta possivel encontrada e que os dadosestao balanceados

In [0]:
display(df.select('avg_glucose_level'))

avg_glucose_level 228.69 202.21 105.92 171.23 174.12 186.21 70.09 94.39 76.15 58.57 80.43 120.46 104.51 219.84 214.09 167.41 191.61 221.29 89.22 217.08 193.94 233.29 228.7 208.3 102.87 104.12 100.98 189.84 195.23 211.78 212.08 83.41 196.92 252.72 84.2 84.03 219.72 74.63 92.62 60.91 78.03 71.22 144.9 90.9 213.03 243.58 109.78 107.26 99.33 58.09 89.96 75.32 127.29 124.13 197.54 196.71 59.32 237.75 194.99 180.93 185.17 74.9 61.94 93.72 104.72 113.01 221.58 104.47 86.23 72.67 62.57 179.12 116.55 228.56 96.59 66.72 240.09 110.85 165.31 143.43 96.16 101.45 88.92 79.79 71.2 96.97 111.81 59.35 86.94 98.55 226.98 72.81 68.02 68.56 64.14 235.63 76.34 240.59 78.92 82.81 74.1 190.32 231.61 78.7 110.52 82.3 73.18 191.82 93.05 64.17 129.98 68.53 106.41 224.1 82.1 216.94 76.11 72.96 82.28 105.22 59.86 62.55 259.63 249.31 219.91 131.41 93.13 73.54 200.59 62.48 190.14 130.54 182.99 101.45 206.09 263.32 140.1 207.28 194.37 199.2 103.68 116.44 70.28 72.17 221.79 151.16 68.43 67.29 67.41 239.07 169.67 223.83 76.57 77.82 92.98 231.56 102.16 221.89 195.71 74.02 57.92 81.76 70.37 203.87 89.13 133.19 162.23 75.02 98.02 91.54 111.98 60.94 97.43 91.02 60.67 185.49 64.44 56.11 110.66 213.22 78.18 111.04 76.13 81.32 112.24 137.3 116.69 215.94 86.62 209.86 205.77 112.16 118.03 271.74 97.92 200.62 242.52 93.81 114.77 74.96 167.13 175.29 90.0 109.47 87.85 106.74 80.13 90.19 134.8 106.58 83.24 103.43 208.65 74.11 95.84 205.33 116.1 210.4 199.86 219.73 250.89 72.73 205.35 70.3 104.86 216.58 184.4 60.98 97.76 66.03 81.95 97.73 70.94 199.84 218.46 211.06 86.3 162.14 88.2 91.92 76.46 197.28 233.94 247.51 84.96 57.93 78.8 87.8 78.81 95.12 87.96 110.89 69.04 161.28 210.95 77.59 243.53 77.67 205.84 77.08 57.08 162.96 73.5 95.04 85.37 84.62 82.67 57.33 67.84 75.7 129.54 60.22 198.21 109.82 60.84 94.61 97.49 206.72 214.45 82.9 103.26 55.78 73.74 149.75 82.34 62.6 94.09 55.42 82.18 117.92 114.84 79.17 110.63 190.7 79.81 113.63 141.24 56.96 203.04 94.3 81.51 137.74 81.84 242.3 102.64 146.01 75.42 220.49 61.45 88.19 115.22 97.93 85.6 79.83 218.46 62.2 69.15 127.71 216.7 71.89 103.56 79.03 69.94 90.12 95.1 234.58 235.85 78.05 89.44 111.38 243.5 182.2 229.92 82.24 89.84 90.69 84.38 215.6 91.61 138.16 75.23 75.18 106.23 239.64 58.95 99.92 116.23 118.82 66.3 101.41 200.28 136.8 96.57 66.42 98.09 205.23 86.78 126.82 78.4 63.69 209.58 89.86 110.99 83.82 137.96 210.78 118.89 72.55 107.41 93.25 74.91 80.05 99.8 91.97 64.4 73.02 88.11 138.29 72.36 68.76 106.22 101.52 109.51 85.84 110.15 91.72 251.6 123.1 97.4 68.34 80.81 120.31 83.51 74.09 91.6 87.52 213.37 75.39 122.41 223.36 84.59 70.16 107.52 80.88 93.48 70.21 86.67 101.37 105.22 70.31 64.06 178.29 81.6 147.48 86.86 71.16 73.2 203.81 82.0 97.55 94.29 75.22 90.07 73.73 79.58 77.88 205.26 58.63 139.9 211.03 72.79 91.65 79.92 225.47 94.48 82.34 180.63 92.48 116.66 112.09 130.34 82.59 123.87 98.7 227.1 55.39 66.07 104.21 145.15 68.19 97.53 122.31 83.51 86.46 112.06 92.14 201.76 74.44 95.93 87.51 72.35 113.47 103.78 115.43 95.59 101.52 170.05 217.3 196.01 92.21 100.09 93.88 75.05 70.08 74.2 69.01 96.17 184.15 87.24 198.69 82.58 96.88 73.32 94.59 132.08 98.99 104.48 98.01 79.14 89.32 65.24 80.98 186.17 69.24 111.19 93.28 122.19 129.53 114.53 93.73 77.49 65.38 94.18 118.7 96.29 74.64 96.99 100.61 97.08 84.79 78.75 82.93 88.78 122.1 74.81 183.45 114.94 104.38 102.91 108.79 73.31 68.52 89.77 102.64 70.92 86.11 129.16 113.85 79.35 80.79 114.01 93.74 90.65 98.41 127.18 138.06 89.44 105.61 155.43 80.83 80.19 60.34 79.3 98.44 210.48 70.34 122.22 75.29 100.35 193.83 126.96 183.34 247.69 76.2 71.43 93.36 61.29 103.62 100.52 191.47 67.06 144.1 58.26 239.82 73.71 88.53 114.79 113.8 71.91 65.3 189.57 114.25 81.26 87.88 112.17 99.97 207.58 109.16 74.46 118.51 107.47 182.86 59.17 215.64 163.7 196.36 139.77 141.23 89.99 108.53 63.61 85.54 100.83 59.82 105.95 188.11 71.15 60.53 205.5 204.86 115.03 79.87 97.89 125.26 114.32 92.23 83.51 102.34 144.48 88.39 228.08 93.79 147.14 106.43 88.04 92.2 219.53 75.56 158.31 111.43 219.97 148.91 69.4 68.34 58.3

Pelo grafico Boxplot se nota que o nivel medio de glicose : 95

Existem outliers nesse dataset, mas nao devem ser desconsiderados,pois demonstram a abstracao da variavel

In [0]:
display(df.select('age'))

age 67.0 61.0 80.0 49.0 79.0 81.0 74.0 69.0 59.0 78.0 81.0 61.0 54.0 78.0 79.0 50.0 64.0 75.0 60.0 57.0 71.0 52.0 79.0 82.0 71.0 80.0 65.0 58.0 69.0 59.0 57.0 42.0 82.0 80.0 48.0 82.0 74.0 72.0 58.0 49.0 78.0 54.0 82.0 63.0 60.0 76.0 75.0 58.0 81.0 39.0 76.0 78.0 79.0 77.0 63.0 63.0 82.0 78.0 73.0 54.0 56.0 80.0 67.0 45.0 75.0 78.0 70.0 76.0 59.0 80.0 76.0 67.0 66.0 63.0 52.0 80.0 80.0 79.0 51.0 43.0 59.0 66.0 79.0 68.0 58.0 54.0 61.0 70.0 47.0 74.0 79.0 81.0 57.0 80.0 45.0 78.0 70.0 58.0 57.0 69.0 64.0 77.0 74.0 81.0 57.0 58.0 50.0 54.0 79.0 53.0 79.0 80.0 76.0 45.0 68.0 71.0 61.0 74.0 38.0 77.0 58.0 53.0 80.0 56.0 72.0 72.0 78.0 80.0 82.0 75.0 73.0 78.0 69.0 38.0 68.0 71.0 76.0 76.0 76.0 75.0 82.0 59.0 74.0 69.0 72.0 66.0 65.0 78.0 57.0 70.0 79.0 68.0 80.0 68.0 55.0 77.0 50.0 57.0 71.0 81.0 76.0 71.0 1.32 78.0 70.0 78.0 56.0 79.0 79.0 81.0 64.0 79.0 60.0 80.0 78.0 72.0 80.0 78.0 80.0 81.0 46.0 59.0 32.0 77.0 61.0 63.0 63.0 82.0 82.0 61.0 65.0 51.0 59.0 68.0 72.0 59.0 70.0 69.0 79.0 74.0 74.0 80.0 77.0 78.0 54.0 78.0 81.0 78.0 78.0 63.0 39.0 51.0 63.0 48.0 81.0 79.0 78.0 55.0 76.0 73.0 81.0 79.0 63.0 81.0 49.0 57.0 81.0 74.0 39.0 80.0 81.0 79.0 73.0 77.0 78.0 68.0 57.0 77.0 51.0 60.0 66.0 57.0 68.0 68.0 57.0 14.0 75.0 71.0 78.0 3.0 58.0 8.0 70.0 14.0 47.0 52.0 75.0 32.0 74.0 79.0 79.0 37.0 37.0 40.0 35.0 20.0 42.0 44.0 79.0 65.0 57.0 49.0 71.0 59.0 25.0 67.0 38.0 54.0 70.0 27.0 47.0 58.0 3.0 58.0 14.0 32.0 23.0 55.0 17.0 59.0 13.0 4.0 16.0 67.0 22.0 45.0 66.0 58.0 69.0 58.0 53.0 78.0 44.0 66.0 27.0 43.0 13.0 51.0 30.0 46.0 43.0 79.0 27.0 38.0 78.0 57.0 61.0 37.0 48.0 29.0 22.0 11.0 22.0 53.0 27.0 76.0 57.0 8.0 21.0 18.0 78.0 78.0 33.0 52.0 51.0 11.0 48.0 82.0 54.0 49.0 24.0 37.0 34.0 64.0 66.0 23.0 68.0 55.0 25.0 48.0 79.0 47.0 20.0 60.0 32.0 66.0 43.0 36.0 52.0 29.0 76.0 39.0 76.0 0.64 34.0 72.0 43.0 57.0 44.0 4.0 44.0 18.0 41.0 23.0 53.0 56.0 0.88 25.0 14.0 53.0 4.0 39.0 53.0 5.0 4.0 37.0 70.0 24.0 55.0 70.0 34.0 26.0 39.0 38.0 20.0 17.0 60.0 75.0 31.0 58.0 5.0 13.0 22.0 52.0 61.0 78.0 57.0 21.0 22.0 80.0 7.0 64.0 64.0 38.0 12.0 33.0 3.0 37.0 22.0 24.0 75.0 12.0 63.0 52.0 68.0 5.0 56.0 43.0 40.0 53.0 52.0 31.0 63.0 69.0 64.0 64.0 62.0 2.0 53.0 65.0 22.0 26.0 53.0 20.0 63.0 13.0 21.0 65.0 65.0 42.0 72.0 13.0 29.0 59.0 42.0 13.0 75.0 33.0 43.0 11.0 7.0 16.0 44.0 78.0 25.0 69.0 61.0 43.0 38.0 68.0 26.0 18.0 43.0 43.0 32.0 81.0 25.0 61.0 66.0 54.0 11.0 53.0 78.0 9.0 48.0 31.0 20.0 21.0 42.0 35.0 53.0 60.0 74.0 23.0 55.0 71.0 5.0 14.0 15.0 66.0 56.0 60.0 57.0 29.0 76.0 52.0 42.0 71.0 64.0 39.0 46.0 51.0 31.0 22.0 41.0 28.0 44.0 68.0 49.0 10.0 31.0 67.0 40.0 70.0 38.0 5.0 54.0 30.0 23.0 31.0 72.0 23.0 25.0 51.0 20.0 12.0 57.0 65.0 16.0 43.0 30.0 29.0 54.0 57.0 42.0 18.0 9.0 26.0 40.0 77.0 15.0 76.0 52.0 55.0 5.0 55.0 36.0 46.0 5.0 39.0 12.0 67.0 1.8 72.0 0.32 64.0 27.0 25.0 81.0 32.0 57.0 52.0 54.0 36.0 18.0 45.0 29.0 51.0 23.0 54.0 54.0 18.0 26.0 72.0 5.0 65.0 42.0 47.0 24.0 57.0 51.0 48.0 56.0 30.0 63.0 46.0 53.0 1.08 15.0 59.0 24.0 53.0 39.0 23.0 31.0 5.0 44.0 1.8 40.0 7.0 56.0 42.0 48.0 82.0 52.0 42.0 73.0 55.0 50.0 43.0 33.0 80.0 33.0 74.0 65.0 40.0 52.0 66.0 20.0 51.0 60.0 59.0 58.0 42.0 28.0 57.0 66.0 56.0 48.0 37.0 27.0 15.0 61.0 80.0 43.0 53.0 70.0 58.0 49.0 65.0 54.0 74.0 56.0 79.0 71.0 28.0 9.0 4.0 36.0 78.0 52.0 80.0 28.0 38.0 56.0 75.0 76.0 10.0 48.0 40.0 82.0 61.0 17.0 58.0 12.0 53.0 26.0 48.0 25.0 27.0 31.0 16.0 20.0 2.0 1.8 28.0 32.0 35.0 52.0 55.0 42.0 52.0 18.0 13.0 45.0 29.0 12.0 37.0 30.0 44.0 79.0 38.0 45.0 66.0 19.0 49.0 66.0 47.0 59.0 23.0 79.0 37.0 78.0 70.0 34.0 28.0 29.0 13.0 22.0 56.0 81.0 28.0 9.0 50.0 70.0 30.0 51.0 24.0 17.0 43.0 9.0 50.0 49.0 6.0 30.0 73.0 20.0 42.0 79.0 55.0 27.0 71.0 9.0 23.0 71.0 56.0 31.0 64.0 46.0 55.0 2.0 58.0 78.0 1.16 35.0 64.0 18.0 19.0 47.0 54.0 39.0 37.0 17.0 20.0 31.0 56.0 31.0 82.0 33.0 42.0 20.0 22.0 61.0 31.0 31.0 17.0 30.0 69.0 63.0 19.0 7.0 51.0 38.0 2.0 58.0 69.0 64.0 31.0 59.0 54.0 24.0 19.0 17.0 29.0 42.0 31.0 29.0 23.0 25.0 14.0 66.0 77.0 79.0 52.0 41.0 31.0 39.

Nessa variavel pode-se afirmar que sua mediana : 52 anos

In [0]:
df.groupby('work_type').count().show()

+-------------+-----+
|    work_type|count|
+-------------+-----+
| Never_worked|   22|
|Self-employed|  819|
|      Private| 2925|
|     children|  687|
|     Govt_job|  657|
+-------------+-----+



Existem 5 tipos de classes para essa variavel

A classe que mais possui instancias : Private 2925

In [0]:
print(f'Tipo de variavel BMI : {df.select("bmi").dtypes}')
print(f'Tipo de variavel SMOKING_STATUS : {df.select("smoking_status").dtypes}')

Tipo de variavel BMI : [('bmi', 'string')]
Tipo de variavel SMOKING_STATUS : [('smoking_status', 'string')]


In [0]:
df.groupby('smoking_status').count().show()

+---------------+-----+
| smoking_status|count|
+---------------+-----+
|         smokes|  789|
|        Unknown| 1544|
|   never smoked| 1892|
|formerly smoked|  885|
+---------------+-----+



Verifica-se que existem 4 classes nessa variavel

In [0]:
#agrupando os dados
display(df.groupby('smoking_status','stroke').count().sort("count",ascending=True).show())

+---------------+------+-----+
| smoking_status|stroke|count|
+---------------+------+-----+
|         smokes|     1|   42|
|        Unknown|     1|   47|
|formerly smoked|     1|   70|
|   never smoked|     1|   90|
|         smokes|     0|  747|
|formerly smoked|     0|  815|
|        Unknown|     0| 1497|
|   never smoked|     0| 1802|
+---------------+------+-----+



Nessa verificao, nota-se que existe maior quantidade de individuos qued nunca fumaram e tiveram AVC

In [0]:
#agrupando os dados
df.groupby('stroke','hypertension').count().sort("count",ascending=True).show()


+------+------------+-----+
|stroke|hypertension|count|
+------+------------+-----+
|     1|           1|   66|
|     1|           0|  183|
|     0|           1|  432|
|     0|           0| 4429|
+------+------------+-----+



Nesse grafico, verifica-se a alta incidencia de AVC em individuos com hypertensao

#Realizando a transformacao dos dados categoricos para a analise

In [0]:
dataset_filtrado=df.filter((df['bmi'] != 'N/A') & (df['smoking_status'] > 'Unknown'))

In [0]:
# aplicando a transformação dos dados categóricos
from pyspark.ml.feature import VectorAssembler, OneHotEncoder, StringIndexer

In [0]:
#define a transformação para a variável "gender"
stringIndexer_gender=StringIndexer(inputCol="gender", outputCol="gender_encoded")  #label encoding
encoder_gender = OneHotEncoder(dropLast=False, inputCol="gender_encoded", outputCol="genderVec") #one-hot encoding

In [0]:
#define a transformação para a variável "ever_married"
stringIndexer_married=StringIndexer(inputCol="ever_married", outputCol="ever_married_encoded") #label encoding
encoder_married = OneHotEncoder(dropLast=False, inputCol="ever_married_encoded", outputCol="marriedVec") #one-hot encoding

In [0]:
#define a transformação para a variável "work_type"
stringIndexer_work=StringIndexer(inputCol="work_type", outputCol="work_type_encoded")  #label encoding
encoder_work = OneHotEncoder(dropLast=False, inputCol="work_type_encoded", outputCol="workVec") #one-hot encoding

In [0]:
#define a transformação para a variável "Residence_type"
stringIndexer_residence=StringIndexer(inputCol="Residence_type", outputCol="Residence_type_encoded")  #label encoding
encoder_residence = OneHotEncoder(dropLast=False, inputCol="Residence_type_encoded", outputCol="residenceVec") #one-hot encoding

In [0]:
#define a transformação para a variável "smoking_status"
stringIndexer_smoking=StringIndexer(inputCol="smoking_status", outputCol="smoking_status_encoded")  #define o objeto
encoder_smoking = OneHotEncoder(dropLast=False, inputCol="smoking_status_encoded", outputCol="smokingVec")#one-hot encoding

In [0]:
#define a construção do vetor de entrada
colunas_entrada=['age','hypertension', 'heart_disease','avg_glucose_level','genderVec','marriedVec','workVec','residenceVec','smokingVec']
vetor_entrada = VectorAssembler(inputCols=colunas_entrada,outputCol='features')

In [0]:
#define a sequencia de transformações para o pipeline
sequencia_transformacoes=[stringIndexer_gender,stringIndexer_married,stringIndexer_work,stringIndexer_residence,stringIndexer_smoking,encoder_gender,encoder_married,encoder_work,encoder_residence,encoder_smoking,vetor_entrada]

In [0]:
from pyspark.ml import Pipeline
# Aplicando o pipeline
pipeline = Pipeline(stages=sequencia_transformacoes)
pipelineModel = pipeline.fit(dataset_filtrado)
model = pipelineModel.transform(dataset_filtrado)

In [0]:
#mostrando parte dos dados para entrada
model.select('age','gender','genderVec','ever_married','marriedVec','features').show()

+----+------+-------------+------------+-------------+--------------------+
| age|gender|    genderVec|ever_married|   marriedVec|            features|
+----+------+-------------+------------+-------------+--------------------+
|67.0|  Male|(3,[1],[1.0])|         Yes|(2,[0],[1.0])|(19,[0,2,3,5,7,9,...|
|80.0|  Male|(3,[1],[1.0])|         Yes|(2,[0],[1.0])|(19,[0,2,3,5,7,9,...|
|49.0|Female|(3,[0],[1.0])|         Yes|(2,[0],[1.0])|(19,[0,3,4,7,9,14...|
|79.0|Female|(3,[0],[1.0])|         Yes|(2,[0],[1.0])|(19,[0,1,3,4,7,10...|
|81.0|  Male|(3,[1],[1.0])|         Yes|(2,[0],[1.0])|(19,[0,3,5,7,9,14...|
|74.0|  Male|(3,[1],[1.0])|         Yes|(2,[0],[1.0])|(19,[0,1,2,3,5,7,...|
|69.0|Female|(3,[0],[1.0])|          No|(2,[1],[1.0])|(19,[0,3,4,8,9,14...|
|81.0|Female|(3,[0],[1.0])|         Yes|(2,[0],[1.0])|(19,[0,1,3,4,7,9,...|
|61.0|Female|(3,[0],[1.0])|         Yes|(2,[0],[1.0])|(19,[0,2,3,4,7,11...|
|54.0|Female|(3,[0],[1.0])|         Yes|(2,[0],[1.0])|(19,[0,3,4,7,9,14...|
|79.0|Female

In [0]:
#dividindo o dataset entre teste e treinamento
train_data, test_data = model.randomSplit([.8,.2],seed=1)

In [0]:
#define o modelo de regrssão logística
from pyspark.ml.classification import LogisticRegression
from pyspark.ml.evaluation import MulticlassClassificationEvaluator

#instancia o objeto para a regressão logística
lr = LogisticRegression(labelCol="stroke",featuresCol="features", maxIter=100, regParam=0.3, )

# treina o modelo
linearModel = lr.fit(train_data)

In [0]:
#realiza a previsão utilizando o modelo de regressão logística
previsao_regressao = linearModel.transform(test_data)

In [0]:
#avaliando a classificação realizada pela regressão logística
acc_evaluator = MulticlassClassificationEvaluator(labelCol="stroke", predictionCol="prediction", metricName="accuracy")
acuracia_regressao = acc_evaluator.evaluate(previsao_regressao)
print('Regressão Logística: {0:2.2f}%'.format(acuracia_regressao*100))

Regressão Logística: 94.31%
